restapi.go

```diff
@@ -2566,3 +2576,222 @@ func (s *Session) FollowupMessageEdit(appID string, interaction *Interaction, me
 func (s *Session) FollowupMessageDelete(appID string, interaction *Interaction, messageID string) error {
 	return s.WebhookMessageDelete(appID, interaction.Token, messageID)
 }
+
+////////////////////////////////
+//////////  THREADS  //////////
+//////////////////////////////
+
+// StartThreadWithMessage creates a new public thread from an existing message.
+// When called on a ChannelTypeGuildText channel, creates a ChannelTypeGuildPublicThread.
+// When called on a ChannelTypeGuildNews channel, creates a ChannelTypeGuildNewsThread.
+// The id of the created thread will be the same as the id of the message,
+// and as such a message can only have a single thread created from it.
+//
+// POST /channels/{channel.id}/messages/{message.id}/threads
+func (s *Session) StartThreadWithMessage(channelID, messageID string, data *ThreadCreateData) (st *Channel, err error) {
+	body, err := s.RequestWithBucketID("POST", EndpointChannelMessageThreads(channelID, messageID), data, EndpointChannelMessageThreads("", ""))
```

Comment:
Given the following passage from the rate limits documentation:
> Additionally, rate limits take into account major parameters in the URL. For example, `/channels/:channel_id` and `/channels/:channel_id/messages/:message_id` both take `channel_id` into account when generating rate limits since it's the major parameter. Currently, the only major parameters are `channel_id`, `guild_id`, and `webhook_id + webhook_token`.

It seems to me like the bucket ID should include the channel ID here, since each different channel will be in a different bucket. I think that this also applies to `StartThreadWithoutMessage`, `JoinThread`, `AddUserToThread`, `LeaveThread`, `RemoveUserFromThread`, `ThreadMembers`, `PublicArchivedThreads`, `PrivateArchivedThreads`, and `JoinedPrivateArchivedThreads` (`ThreadEditComplex` and `ListActiveThreads` already do this).